ddtrace/internal/sampling.py

```diff
@@ -208,44 +210,62 @@ def apply_span_sampling_tags(self, span):
 
 def get_span_sampling_rules():
     # type: () -> List[SpanSamplingRule]
-    json_rules_raw = os.getenv("DD_SPAN_SAMPLING_RULES")
-    if json_rules_raw is None:
-        return []
+    env_json_rules = os.getenv("DD_SPAN_SAMPLING_RULES")
+    file_json_rules = os.getenv("DD_SPAN_SAMPLING_RULES_FILE")
+    if env_json_rules and file_json_rules:
+        log.warning(
+            (
+                "DD_SPAN_SAMPLING_RULES and DD_SPAN_SAMPLING_RULES_FILE detected. "
+                "Defaulting to DD_SPAN_SAMPLING_RULES value."
+            )
+        )
+        raw_json_rules = env_json_rules
+    elif env_json_rules:
+        raw_json_rules = env_json_rules
+    elif file_json_rules:
+        with open(file_json_rules) as f:
+            raw_json_rules = f.read()
+    # No rules specified
     else:
-        sampling_rules = []
+        return []
+
+    try:
+        json_rules = json.loads(raw_json_rules)
+        if not isinstance(json_rules, list):
+            raise TypeError("DD_SPAN_SAMPLING_RULES is not list, got %r" % json_rules)
+    except JSONDecodeError:
+        raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES=%r" % env_json_rules)
+
+    sampling_rules = []
+    for rule in json_rules:
```

Comment:
this whole thing should be written with a schema library at this point, pydantic or the like 😢

Reply:
Pydantic looks cool, I'd be happy to give that a go but wondering if there's a reason we don't use it anywhere else? Since it uses Python 3.6 type hinting, could that cause issues for 2.7?

Reply:
Will take a look at json schema validator